prov/verbs/src/verbs_info.c

```diff
@@ -1457,21 +1457,50 @@ static int fi_ibv_get_match_infos(uint32_t version, const char *node,
 	if (ret)
 		return ret;
 
-	if (hints && (hints->addr_format == FI_ADDR_IB_UD)) {
+	/* Check if the user requested to support DGRAM EP type only */
+	if (hints && ((hints->addr_format == FI_ADDR_IB_UD) ||
+		      (hints->ep_attr && (hints->ep_attr->type == FI_EP_DGRAM)))) {
 		/* This is case when only IB UD addresses are passed */
 		ret = fi_ibv_handle_ib_ud_addr(node, service, flags, info);
-		if (ret)
+		if (ret) {
+			VERBS_INFO(FI_LOG_CORE,
+				   "Handling of the IB UD address fails - %d, "
+				   "support of this was requested thru the passed hints\n",
+				   ret);
 			fi_freeinfo(*info);
+		}
+		return ret;
+	}
+
+	/* Check if the user requested to support MSG EP type only */
+	if (hints && ((hints->addr_format == FI_SOCKADDR) ||
+		      (hints->addr_format == FI_SOCKADDR_IN) ||
+		      (hints->addr_format == FI_SOCKADDR_IN6) ||
+		      (hints->addr_format == FI_SOCKADDR_IB) ||
+		      (hints->ep_attr && (hints->ep_attr->type == FI_EP_MSG)))) {
```

Comment:
Can you create a couple helper functions to the MSG/DGRAM EP checks?

Reply:
Done